### PR TITLE
test: Add and use uint256_to_string + improve some hash/string conversions

### PR DESCRIPTION
**Pull request by xdustinface**

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK

File: test/functional/test_framework/messages.py [hunk 1]
```diff
@@ -106,6 +106,10 @@ def uint256_from_str(s):
     return r
 
 
+def uint256_to_string(uint256):
+    return '%064x' % uint256
+
+
```
Comment (PastaPastaPasta) [inline test/functional/test_framework/messages.py:112]: Why should this be in messages.py? why not util.py?

Comment (xdustinface) [reply test/functional/test_framework/messages.py:112]: I just put it below `uint256_from_string`

Comment (PastaPastaPasta) [reply test/functional/test_framework/messages.py:112]: Oh, that makes sense :D

**Comment by PastaPastaPasta** [comment]
What if we add a linter for `"%064x" %`?

**Comment by xdustinface** [comment]
> What if we add a linter for `"%064x" %`?

If i didn't miss something it seems like thats probably not worth the effort if you were talking about modifying `lint-python.sh` 🙈  See https://flake8.pycqa.org/en/latest/plugin-development/index.html

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see b91dc78007

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK, should probably change title, as this is a little more than that now

**Pull request merged**